cmd/trace-agent/config/config.go

```diff
@@ -138,7 +138,11 @@ func applyDatadogConfig(c *config.AgentConfig) error {
 		c.StatsdPort = coreconfig.Datadog.GetInt("dogstatsd_port")
 	}
 
-	c.Endpoints[0].Host = coreconfig.GetMainEndpoint(apiEndpointPrefix, "apm_config.apm_dd_url")
+	if coreconfig.Datadog.GetBool("vector.traces.enabled") && coreconfig.Datadog.IsSet("vector.traces.url") && len(coreconfig.Datadog.GetString("vector.traces.url")) > 0 {
```

Comment:
Shouldn't the user know if `vector.traces.url` happens to be empty? For that reason, I suggest:
```suggestion
	if coreconfig.Datadog.GetBool("vector.traces.enabled") {
	    if host := coreconfig.Datadog.GetString("vector.traces.url"); host == "" {
	        log.Error("vector.traces.enabled but vector.traces.url is empty")
	    } else {
	        c.Endpoints[0].Host = host
	    }
```
Also, you're using `GetBool` straight up, without `IsSet`, so let's do the same for `GetString`.

Reply:
Interesting, I borrowed the `IsSet` before `GetString` from [here](https://github.com/DataDog/datadog-agent/blob/main/pkg/logs/config/config_keys.go#L54-L56) but your are right, I double checked and viper indeed returns "" for a non-existing key.